### PR TITLE
fix(server): Add auth-strategy creation to install-plugin migration s…

### PR DESCRIPTION
**Pull request by morhag90**
Close: #4431 

## PR Details

Add auth-strategy plugin if it does not exist in install-plugin migration script
## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

Test cases: 

Add new services. 
DO NOT install plugins: auth-core / auth-basic/ auth-jwt 
Run the migration script: install-auth-core-plugin with the following command: 
POSTGRESQL_URL=postgres://[user]:[password]@127.0.0.1:5432/app-database npx ts-node install-auth-core-plugin.ts 

Check that every new service you created in the first step has installed plugins: 
auth-core
auth-basic/ auth-jwt  (according to the auth provider type).

### CODE REVIEW

File: packages/amplication-server/migration-scripts/install-auth-core-plugin.ts [hunk 1]
```diff
@@ -90,31 +118,69 @@ async function main() {
...
+            settings: {},
+            version: "latest",
+            resource: {
+              connect: {
+                id: resource.id,
+              },
+            },
+          },
+        });
+        console.log({ AuthStrategyPlugin });
```
Comment (barshimi) [inline packages/amplication-server/migration-scripts/install-auth-core-plugin.ts:180]: remove console and I suggest to wrap it in try/catch

File: packages/amplication-server/migration-scripts/install-auth-core-plugin.ts [hunk 2]
```diff
@@ -82,6 +84,32 @@ async function main() {
...
+  ): boolean {
+    const results = plugins.filter((plugin) => {
+      return (
+        plugin.pluginId ===
+        authPluginNames.find((name) => name === plugin.pluginId)
+      );
+    });
+
+    if (results.length > 0) return true;
+    return false;
```
Comment (tupe12334) [inline packages/amplication-server/migration-scripts/install-auth-core-plugin.ts:111]: Same as above

File: packages/amplication-server/migration-scripts/install-auth-core-plugin.ts [hunk 3]
```diff
@@ -90,31 +118,69 @@ async function main() {
...
+            resource: {
+              connect: {
+                id: resource.id,
+              },
             },
           },
-        },
-      });
+        });
+        console.log({ newPlugin });
```
Comment (tupe12334) [inline packages/amplication-server/migration-scripts/install-auth-core-plugin.ts:149]:
```suggestion
```

File: packages/amplication-server/migration-scripts/install-auth-core-plugin.ts [hunk 4]
```diff
@@ -82,42 +84,108 @@ async function main() {
...
+  function isPluginExist(
+    plugins: PluginInstallation[],
+    authProvider: string
+  ): boolean {
+    const plugin = plugins.find(
+      (plugin) => plugin.pluginId.trim() == authProvider
+    );
+
+    if (plugin) return true;
+    return false;
```
Comment (abrl91) [inline packages/amplication-server/migration-scripts/install-auth-core-plugin.ts:96]: we actually can use the `.some()` method that returns boolean: true if it found the item and false if it didn't find
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

```suggestion
    return plugins.some(
      (plugin) => plugin.pluginId.trim() == authProvider
    );
```

Comment (abrl91) [reply packages/amplication-server/migration-scripts/install-auth-core-plugin.ts:96]: But it's just a suggestion...we can leave it like that

**Review by abrl91 — APPROVED** [review]

**Pull request merged**